src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -421,7 +422,13 @@ private SerializableRecordingDescriptor startRecording(StartRecordingRequest req
                         localStorageTemplateService.addTemplate(
                                 new ByteArrayInputStream(
                                         req.template.getBytes(StandardCharsets.UTF_8)));
-                events = localStorageTemplateService.getEvents(template).orElseThrow();
+                events =
+                        localStorageTemplateService
+                                .getEvents(template)
+                                .orElseThrow(
+                                        () ->
+                                                new InvalidEventTemplateException(
+                                                        "Falied to retrieve recording events"));
```

Comment:
It might be nice to include the requested template's name in these exception messages.